### PR TITLE
Tech Debt - End to end - Initial implementation

### PR DESCRIPTION
**Pull request by tgandrews**
### What is the context of this PR?
Added initial version of an end to end test run across author and proves we are still publishing the same format.

1. Renames the previous end to end tests to be called integration tests.
1. Runs tests against author running in container not on travis directly. 

### How to review 
1. Check if this structure makes sense.
1. Ensure that you can locally run `yarn test:e2e`

### CODE REVIEW

**Comment by tgandrews** [comment]
[Author end to end test](https://trello.com/c/TLUrpWy6/497-author-end-to-end-test)

File: cypress/e2e/e2e_spec.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+          }
+        ]
+      }
+    });
+  });
+
+  it("Publishes successfully", () => {
+    cy.request(`http://localhost:9000/publish/${questionnaireId}`).then(
+      response => {
+        // cy.writeFile(
```
[the remark this answers is not in the document]
Comment (tgandrews) [reply cypress/e2e/e2e_spec.js]: It is intentionally left so you can get the new publisher output if changes are made to the test suite.

[the remark this answers is not in the document]
Comment (adammahmood) [reply cypress/e2e/e2e_spec.js]: Can we either annotate the comments to explain this or set a flag somewhere to switch between the two if needed?

File: cypress/builders/questionnaire.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import { testId } from "../utils";
+
+const checkIsOnDesignPage = () => cy.hash().should("match", /\/design$/);
+
+const updateDetails = ({ title }) => {
+  cy.get(`[data-testid="questionnaire-settings-modal"]`).within(() => {
```
[the remark this answers is not in the document]
Comment (tgandrews) [reply cypress/builders/questionnaire.js]: I'll make this change one the eslint changes are merged.

File: cypress/e2e/e2e_spec.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (tgandrews) [reply cypress/e2e/e2e_spec.js]: I've added an extra comment but agree that we could look into a flag later on. The diff is pretty good but if you've just added a feature (or more tests) then its easier to just save the complete new output.

File: cypress/builders/questionnaire.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (adammahmood) [reply cypress/builders/questionnaire.js]: The eslint has just been merged.

**Comment by adammahmood** [comment]
LGTM - Remember to squash commits.

**Pull request merged**